src/main/java/org/dependencytrack/model/WorkflowState.java

```diff
@@ -0,0 +1,136 @@
+package org.dependencytrack.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import javax.jdo.annotations.Column;
+import javax.jdo.annotations.FetchGroup;
+import javax.jdo.annotations.FetchGroups;
+import javax.jdo.annotations.IdGeneratorStrategy;
+import javax.jdo.annotations.PersistenceCapable;
+import javax.jdo.annotations.Persistent;
+import javax.jdo.annotations.PrimaryKey;
+import javax.validation.constraints.NotNull;
+import java.io.Serializable;
+import java.util.Date;
+import java.util.UUID;
+
+@PersistenceCapable(table= "WORKFLOW_STATE")
+@FetchGroups({
+        @FetchGroup(name = "ALL", members = {
+                @Persistent(name = "id"),
+                @Persistent(name = "parent"),
+                @Persistent(name = "token"),
+                @Persistent(name = "updatedAt"),
+                @Persistent(name = "startedAt"),
+                @Persistent(name = "step"),
+                @Persistent(name = "status")
+        })
+})
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class WorkflowState implements Serializable {
+
+    public enum FetchGroup {
+        ALL
+    }
+
+    @PrimaryKey
+    @Persistent(valueStrategy = IdGeneratorStrategy.NATIVE)
+    private long id;
+
+    // null allowed because first step will have no parent and will be the first step of recursion
```

Comment:
True, marking this an non-nullable in the schema draft in https://github.com/DependencyTrack/hyades/issues/664 was a mistake. Just corrected that.